### ISSUE TITLE
Shadow mapping breaks visuals in minetest-mod-drawers

### ISSUE THREAD

**Issue by neoh4x0r** [minetest/minetest#11326]
##### Minetest version
Commit: c47313db65f968559711ac1b505ef341a9872017
```
$ minetest --version
Minetest 5.5.0-dev-c47313db6 (Linux)
Using Irrlicht 1.9.0mt2
Using Lua 5.1.5
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
USE_FREETYPE=1
STATIC_SHAREDIR="/usr/local/share/minetest"
STATIC_LOCALEDIR="/usr/local/share/locale"
```

##### OS
```
          Distribution: Debian 
               Version: 10 (debian version: 10.9)
     Support Lifecycle: N\A
              Codename: 10 (buster) 
              Fullname: Debian GNU/Linux 10 (buster) 

           Kernel Name: Linux
        Kernel Release: 5.10.0-0.bpo.5-amd64
        Kernel Version: #1 SMP Debian 5.10.24-1~bpo10+1 (2021-03-29)
  Machine Architecture: x86_64
      Operating System: GNU/Linux
```
#####  CPU
```
Intel(R) Core(TM) i7-2600 CPU @ 3.40GHz / Quad Core / with Hyper-Threadnig
```

##### GPU
```
$ glxinfo | grep -i opengl
OpenGL vendor string: NVIDIA Corporation
OpenGL renderer string: GeForce GTX 550 Ti/PCIe/SSE2
OpenGL core profile version string: 4.6.0 NVIDIA 390.143
OpenGL core profile shading language version string: 4.60 NVIDIA
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
OpenGL version string: 4.6.0 NVIDIA 390.143
OpenGL shading language version string: 4.60 NVIDIA
OpenGL context flags: (none)
OpenGL profile mask: (none)
OpenGL ES profile version string: OpenGL ES 3.2 NVIDIA 390.143
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```

##### Summary
The drawer visuals from the drawers mod https://github.com/minetest-mods/drawers are not displayed correctly.

**With Shadows Disabled:**
![drawers_shadows_disaled](https://user-images.githubusercontent.com/7200584/121245068-98792a00-c86d-11eb-9aa9-7fb295e806db.png)

**With Shadows Enabled:**
![drawers_shadows_enabled](https://user-images.githubusercontent.com/7200584/121245088-a038ce80-c86d-11eb-8a1e-5f02a138c989.png)

##### Steps to reproduce
1. Install the drawer mod
1. with shadows enabled
1. Place any drawer
1. Select an item on the hot bar and left-click on an empty drawer to store the item

The drawers visual will be very dark.
If shadows are disable the visual will look as expected.

**Comment by neoh4x0r** [comment]
@0xLiso I'm also thinking that the cause of this bug might also be responsible for strange self-shadows on the player as well as other non-map objects.

**Comment by 0xLiso** [comment]
Yes, It could be ;). 
I'm also working in to the self shadowing using another approach to the bias calculation. But if you think there is another way, please share with me, it will help a lot :)

**Comment by neoh4x0r** [comment]
> Yes, It could be ;).
> I'm also working in to the self shadowing using another approach to the bias calculation. But if you think there is another way, please share with me, it will help a lot :)

I'm still trying to figure out / understand the code (how it all works) -- so I don't have an alternative solution or and advice at the moment.

**Comment by VanessaE** [comment]
A similar effect can be seen with signs_lib signs -- a sign on the sunlit side of a building (and facing the sun) looks fine, while on signs on the shaded side (facing away from the sun), the text is much too dark.  Signs facing north or south are a little too dark as well, but not to the same degree as above.

**Comment by x2048** [comment]
@neoh4x0r The problem should be fixed in #11747 now, including colored shadows. Would you be able to check it? Thank you.

**Comment by x2048** [comment]
@sfan5 This bug is a real issue, I am addressing this in #11747 now, can we remove the 'Possible Close' label?

**Comment by neoh4x0r** [comment]
**Ok, so I checked out PR #11747**
```bash
$ mkdir minetest && cd minetest
$ git clone https://github.com/minetest/minetest .
$ git fetch origin pull/11747/head:shadows_entities
$ git checkout shadows_entities
```

It looks like issue with the drawer visuals being overly dark has been fixed by PR #11747.

However, dropped entities and the player (while being properly colored and not dark, when in direct light) they have
strange banding/jaggedness in the shadows -- which become more apparent with rotatation towards/away from the light source (sun).

It sounds like issue #11326 can be closed (since it was related to the drawer visuals).
A new issue might need to be opened for the shadow rotation problem (as described above).

**These are the results**
![2021-11-04_15-09](https://user-images.githubusercontent.com/7200584/140405254-ebc4d647-6b12-444c-86ee-b99a3a2998f2.png)

![2021-11-04_15-09_1](https://user-images.githubusercontent.com/7200584/140405283-9a0014b3-7699-4199-b36b-ea7951e4b0af.png)

![2021-11-04_15-10](https://user-images.githubusercontent.com/7200584/140405304-09c34834-9f81-43af-a25c-9c09ecb1059c.png)

**Comment by x2048** [comment]
Do you still observe jagged self-shadowing with medium og higher SM settings? The two lowest settings use very low texture resolution and are known not to be really playable. I am aiming at fixing entities in medium and higher and tune the two lowest presets separately.

**Comment by neoh4x0r** [comment]
> Do you still observe jagged self-shadowing with medium og higher SM settings? The two lowest settings use very low texture resolution and are known not to be really playable. I am aiming at fixing entities in medium and higher and tune the two lowest presets separately.

Yeah turning up the quality (eg. shadow filters and/or poisson filter) removes the jaggies but at the cost of lowering the framerate by 50% or more (say 60 fps going to 30 or 15 fps) -- the more stuff there is to render (with shadows) makes the problem even worse by becoming very laggy..

**Comment by x2048** [comment]
@neoh4x0r I have created #11762 to capture the problem with very low/low preset and will address general entity lighting problems (across all levels) in this bug.
